schism/compiler.ss

```diff
@@ -510,22 +521,11 @@
            (imports (runtime-imports))
            (env (add-intrinsics (intrinsics)
                                 (add-imports imports (empty-env))))
-           (body-env (add-imported
-                      ;; every library implicitly imports itself
-                      (cons (cadr lib) library-imports)
-                      import-envs
-                      env))
-           (body-env (add-top-levels-filter
-                      ;; Filter out names that are exported, since
-                      ;; those have already been included.
-                      (lambda (def)
-                        (let ((name (definition-name def)))
-                          (and name (not (memq name exports)))))
-                      defs
-                      body-env)))
+           (body-env (add-imported library-imports import-envs env))
+           (body-env (add-top-levels defs body-env)))
       (cons exports
```

Comment:
In order to preserve the order of definitions (to detect unbound references), I removed the filter which checked for `export`ed definitions. Is this ok?

Reply:
So we end up with the same definition in the environment twice for imports? I think that's okay.